compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/AnnotationReference.kt

```diff
@@ -86,15 +137,31 @@ public fun AnnotationDescriptor.toAnnotationReference(module: ModuleDescriptor):
 }
 
 @ExperimentalAnvilApi
-public fun AnnotationReference.scope(
-  parameterIndex: Int = 0
-): ClassReference {
-  return when (this) {
-    is Psi ->
-      annotation
-        .findAnnotationArgument<KtClassLiteralExpression>(name = "scope", index = parameterIndex)
-        ?.requireFqName(module)
-        ?.toClassReference(module)
-    is Descriptor -> annotation.scope(module).toClassReference(module)
-  } ?: throw AnvilCompilationException(message = "Couldn't find scope for $fqName.")
+@Suppress("FunctionName")
+public fun AnvilCompilationExceptionAnnotationReference(
+  annotationReference: AnnotationReference,
+  message: String,
+  cause: Throwable? = null
+): AnvilCompilationException = when (annotationReference) {
+  is Psi -> AnvilCompilationException(
+    element = annotationReference.annotation,
+    message = message,
+    cause = cause
+  )
+  is Descriptor -> AnvilCompilationException(
+    annotationDescriptor = annotationReference.annotation,
+    message = message,
+    cause = cause
+  )
+}
+
+private fun replacesIndex(annotationFqName: FqName): Int {
+  return when (annotationFqName) {
+    contributesToFqName -> 1
+    contributesBindingFqName, contributesMultibindingFqName -> 2
+    contributesSubcomponentFqName -> 4
```

Comment:
If these indexes are used anywhere else, I feel like we should make constants for them as well.

Reply:
They aren't used anywhere else. They're only hardcoded here and the values are implied by the order of the parameters in the annotation.